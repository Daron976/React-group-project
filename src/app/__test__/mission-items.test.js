import renderer from 'react-test-renderer';
import MissionItems from '../__mocks__/mission-items';

describe('Mission items', () => {
  const missionArray = [
    {
      id: 1,
      name: 'test',
      description: 'this is a test description',
    },
  ];

  test('mission items renders as expected', async () => {
    const snapMissionItems = renderer.create(
      <MissionItems
        key={missionArray[0].id}
        id={missionArray[0].id}
        name={missionArray[0].name}
        description={missionArray[0].description}
      />,
    )
      .toJSON();
    expect(snapMissionItems).toMatchSnapshot();
  });
});
