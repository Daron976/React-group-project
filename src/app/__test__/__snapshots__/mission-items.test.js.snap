// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mission items mission items renders as expected 1`] = `
<tr>
  <td
    className="td-name"
  >
    test
  </td>
  <td
    className="td-description"
  >
    this is a test description
  </td>
  <td>
    <div
      className="td-status"
      style={
        Object {
          "background": "#6c767c",
        }
      }
    >
      NOT A MEMBER
    </div>
  </td>
  <td
    className="td-btn"
  >
    <button
      className="btn"
      id="join"
      name="join"
      onClick={[Function]}
      style={
        Object {
          "border": "1px solid #7a7a7a",
          "color": "black",
        }
      }
      type="button"
    >
      Join Mission
    </button>
  </td>
</tr>
`;
